packages/atlas-service/src/make-automation-agent-op-request.ts

```diff
@@ -1,3 +1,14 @@
+export type AtlasIndexStats = {
```

Comment:
Sorry to comment on this, i randomly clicked on the pr and it caught my eye, definitely a nit, and a question also for @gribnoysup. 

Isn't this inconsistent with the rest of the design of AtlasService? For the way it is setup for other calls we shouldn't have domain related concepts and types on the atlas service directly, and they should live inside the services that are related to those domains. 

It should all still work with the generic, just the definition of the type doesn't seem to fit the rest of the design. Unless something changed recently.

Reply:
Le Roux is just moving types here, so it's coming from my initial implementation of the automation agent request helper. I think it's fair that this is maybe giving this service too much knowledge about the API, but the API is not very consistent and requires to know more about what commands specifically are being executed if we want to have a single way of doing these two-part agent requests. See [here](https://github.com/mongodb-js/compass/blob/d82bfb543d377c0ddab79defb4a01ced9c39d565/packages/atlas-service/src/make-automation-agent-op-request.ts#L168-L173) for example, certain types of automation agent requests don't return a request id we can wait on, so we need a narrow typing to know what are those cases, the request params and the way response is wrapped in await is also based on the command you're sending, etc.

We can change this to a generic type and move narrower domain specific stuff to rolling indexes, we'd also need to make the assertions we're doing inside the method configurable from outside, or just dismantle this helper in two methods that just know the URL shape for creating these requests and then we re-assemble it into the same logic separately for global writes and rolling indexes. Not very dry, but it's also just two features we need this for right now, so might be worth it to remove the types from here.

Reply:
I actually have a bunch of scattered TODO comments/reminders to talk about this. This type doesn't map well to our own index type at all and I'm not 100% sure where we should do the mapping/assertions.

Reply:
Apologies if i'm misunderstanding the code, but looking at https://github.com/mongodb-js/compass/blob/d82bfb543d377c0ddab79defb4a01ced9c39d565/packages/atlas-service/src/make-automation-agent-op-request.ts#L168-L173, making this behaviour configurable from outside and externalizing the type seems definitely the right thing to do (assuming that we want to keep `AtlasService` as a building block for domain specific services). 

The fact that you need ifs by "opType" there means that we are mixing up responsibilities already, and is not obvious at all what is happening there. It looks also pretty simple to just split that method in two `makeAutomationAgentOpRequest<RT>` and `makeAutomationAgentOpRequestAndWait<RT>` so we don't rely on too many parameters and ifs.

> can definitely blame me for that 

No blame at all!

Reply:
Okay, we'll split it into parts

cc @mabaasit just FYI you'd need to account for that for Global Writes implementation

Reply:
Opened https://github.com/mongodb-js/compass/pull/6305 to make the split